core/src/main/java/com/datastax/oss/driver/internal/core/metadata/MetadataManager.java

```diff
@@ -167,7 +171,31 @@ public void addContactPoints(Set<EndPoint> providedContactPoints) {
    * @see #wasImplicitContactPoint()
    */
   public Set<DefaultNode> getContactPoints() {
-    return contactPoints;
+    return resolvedContactPoints;
+  }
+
+  @Synchronized
```

Comment:
It does the same, but on the field `Lock`, instead of `this`, but what a trip, in my head all the code uses `@Synchronized`, while it is not. switched to method modifier instead.